src/constructs/lambda/lambda.ts

```diff
@@ -43,6 +43,10 @@ export interface GuFunctionProps extends GuDistributable, Omit<FunctionProps, "c
    * accounts.
    */
   withoutFilePrefix?: boolean;
+  /**
+   * TODO
+   */
+  managedDeployment?: boolean;
```

Comment:
👀

Reply:
@akash1810 I've added some documentation here, but I wonder if this parameter should more explicitly be called something like `skipUpload`? What do you think? I'm slightly uncomfortable coupling the name to the functionality coded elsewhere, especially as the `riff-raff.yaml` autogeneration is still "experimental" (IMO it no longer is!).

Reply:
> I wonder if this parameter should more explicitly be called something like `skipUpload`?

👍🏽 this feels better, yes! We've a few `withoutX` props in the library, it's not a formal pattern, but wonder if `withoutArtifactUpload` is an alternative name?

> especially as the `riff-raff.yaml` autogeneration is still "experimental" (IMO it no longer is!).

We've yet to define a process to promote experimental features. Maybe we just move it and create a new major version?

Reply:
Will rename to `withoutArtifactUpload`. 

> Maybe we just move it and create a new major version? This wasn't covered by the [initial ADR](https://github.com/guardian/cdk/blob/main/docs/architecture-decision-records/005-x01-package-structure.md).

This sounds good to me. I think the feature now has enough use for us to say it's not experimental anymore.

Reply:
@kenoir yep! We discussed this recently and will hopefully promote GuRoot to non-experimental soon, as it now has wide usage and proven value.